lib/rspec/request_snapshot/handlers/json.rb

```diff
@@ -42,4 +42,12 @@ def deep_transform_array(hash, key)
   def sort_elements(hash, key)
     hash[key].first.is_a?(Hash) ? hash[key].sort_by! { |e| e.keys.map { |k| e[k] } } : hash[key].sort!
   end
+
+  def handle_dynamic_attribute(hash, key)
+    if dynamic_attributes_with_regex.key?(key)
+      hash[key] = dynamic_attributes_with_regex[key] if dynamic_attributes_with_regex[key].match(hash[key].to_s)
```

Comment:
Use `match?` instead of `match` when `MatchData` is not used.

Reply:
match? is not present in older ruby versions. I think it was added on ruby 2.4 or somewhere in 2.3 versions
That's why I'm using match here